tokenizer_ts/src/bytePairEncode.ts

```diff
@@ -26,16 +26,16 @@ export function bytePairEncode(
     return [ranks.get(mergingBytes[0].toString())!];
   }
 
-  const byteIndicesAndRanks: [number, number][] = [];
+  const byteIndicesAndRanks = new Array<{ index: number; rank: number }>(mergingBytes.length + 1);
```

Comment:
Because everything is known beforehand, I think a better approach is to create a single Int32Array with enough memory for byteIndiciesAndRanks and then create an array of subarrays from that (Int32Array[]). However, I have not measured this yet.